commonmark/src/main/java/org/commonmark/renderer/html/DefaultUrlSanitizer.java

```diff
@@ -0,0 +1,83 @@
+package org.commonmark.renderer.html;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ *
+ * Allows http, https and mailto protocols for url.
+ * Also allows protocol relative urls, and relative urls.
+ * Implementation based on https://github.com/OWASP/java-html-sanitizer/blob/f07e44b034a45d94d6fd010279073c38b6933072/src/main/java/org/owasp/html/FilterUrlByProtocolAttributePolicy.java
+ */
+public class DefaultUrlSanitizer implements UrlSanitizer {
+    private Collection<String> protocols;
+
+    private static final long HTML_SPACE_CHAR_BITMASK =
+            (1L << ' ')
+                    | (1L << '\t')
+                    | (1L << '\n')
+                    | (1L << '\u000c')
+                    | (1L << '\r');
+
+
+    public DefaultUrlSanitizer() {
+        this(new ArrayList<String>() {{
+            add("http");
+            add("https");
+            add("mailto");
+        }});
+    }
+
+    public DefaultUrlSanitizer(Collection<String> protocols) {
+        this.protocols = protocols;
```

Comment:
Let's make a defensive copy of the passed in Collection here using `new HashSet<String>(protocols)`